src/engine/IndexScan.h

```diff
@@ -91,6 +91,12 @@ class IndexScan final : public Operation {
     return numVariables() == target;
   }
 
+  // Full index scans will never be able to fit in the cache on datasets the
+  // size of wikidata, so we don't even need to try and waste performance.
+  [[nodiscard]] bool unlikelyToFitInCache() const override {
```

Comment:
Don'use [[nodiscard]] for such simple functions. You can turn off that clang-tidy check.
Nodiscard only for RAII types etc. and then with an explicit reason please ( I think that is also the sonarcloud policy).

Reply:
Mostly did that to mirror the behaviour of `supportsLimit()`